gaps/gap-16_golem_compose/gap-16_golem_compose.md

```diff
@@ -0,0 +1,177 @@
+```
+
+Override file:
+```
+version: "0.0.1"
+
+payment:
+  budget: 1
+
+payloads:
+  nginx:
+    params:
+      repo: "repo-url"
+    capabilities:
+      - "gpu"
+```
+
+Resulting file:
+```
+version: "0.0.1"
+
+payment:
+  budget: 1
+  driver: "polygon"
+
+payloads:
+  nginx:
+    runtime: "vm"
+    params:
+      image: "image-hash"
+      repo: "repo-url"
+    constraints:
+      "golem.inf.cpu.cores": 2
+    capabilities:
+      - "vpn"
+      - "gpu"
+```
+
+A depth-first approach is used to determine the values which need to be added or updated to existing collections (lists and key-value maps).
+For maps, keys from overriding files have precedence over the base ones.
+In the case of lists, when merging lists from two files, the override values are simply concatenated to the base list. If required, this behaviour can be made configurable (e.g. to enable overriding the entire list instead).
+
+[TODO: describe how these files can be composed (CLI, file system hierarchy)]
+
+#### File schema
+Once the name and the structure of the descriptor file is finalized, its schema can be published to https://www.schemastore.org/json/.
+This way, the YAML language server will provide support for schema validation and completion in IDEs and editors.
+
+## Rationale
+#### File format
+The file format of choice is **YAML**.
+YAML is used in both Docker Compose and Kubernetes, both of which are widely-adopted deployment solutions. As such, it's become a de-facto standard
+1. It's an established standard
+2. It's more flexible compared to alternatives (e.g. TOML)
+3. Users of the solutions mentioned above should find it easy to start using Golem Compose
+
+#### Market strategy
+In its current form, the deployment descriptor **does not** include support for specifying the market strategy which should be used by the requestor. There are two primary reasons for this:
+1. A feasible approach which would allow for specifying/defining market strategies in a language-agnostic matter could not be found in this iteration of the proposal.
+2. Including some form of support for market strategies at an early stage of development would unnecessarily increase complexity.
+Market strategies will be addressed in a future GAP.
+
+#### Inspirations
+- [Docker Compose](https://github.com/compose-spec/compose-spec/blob/master/spec.md)
+- [Terraform](https://www.terraform.io/language)
+- [Kubernetes](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects)
+
+## Backwards Compatibility
+All GAPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The GAP **must** explain how the author proposes to deal with these incompatibilities.
+
+## Test Cases
+Test cases are very useful in summarizing the scope and specifics of a GAP.  If the test suite is too large to reasonably be included inline, then consider adding it as one or more files in `./gaps/gap-draft_title/` directory.
```

Comment:
I would very much recommendadding a couple of "use case examples" of descriptors of applications of various levels of complexity, eg:
- just one node for a VM-based task
- a single-node static website published via Golem VPN and requestor's address/port
- a multi-node load-balanced website with database tier (on separate node, available via Golem VPN)